Content/Items/Jungle/Weapons.ManEaterPot.cs

```diff
@@ -0,0 +1,332 @@
+				float longestDist = (float)Math.Pow(TARGET_RADIUS, 2) + 1;
+
+				foreach (Item item in Main.item)
+				{
+					if (item.active && item.type == ItemID.Heart)
+					{
+						float dist2 = Vector2.DistanceSquared(item.Center, Owner.Center);
+						if (dist2 <= Math.Pow(TARGET_RADIUS, 2) && dist2 < longestDist)
+						{
+							heartTarget = item;
+							longestDist = dist2;
+						}
+					}
+				}
+
+				// If we found a valid heart, go into heart eating mode and stop searching
+				if (heartTarget != null && heartTarget.active)
+				{
+					State = 1;
+					return;
+				}
+			}
+
+			// Else, fall back to targeting an enemy
+			npcTarget = Projectile.TargetClosestNPC(TARGET_RADIUS, true, true);
+
+			// If we found one, go into attack mode
+			if (npcTarget != null && npcTarget.active)
+				State = 2;
+		}
+
+		/// <summary>
+		/// The minion having found a heart, attempts to move towards it, and when close enough 'eats' it,
+		/// giving up if the heart its' trying to eat stops existing
+		/// </summary>
+		public void EatHeart()
+		{
+			// Abort if target is lost
+			if (heartTarget is null || !heartTarget.active || heartTarget.type != ItemID.Heart)
+			{
+				heartTarget ??= null;
+				State = 0;
+				return;
+			}
+
+			Projectile.position += (heartTarget.Center - Projectile.Center) * 0.04f;
+
+			Projectile.position += Projectile.Center.DirectionTo(Owner.Center) * (float)Math.Sin(Projectile.timeLeft * 0.1f + Projectile.minionPos) * 2.5f;
+
+			// If we intersect the heart, consume it
+			if (Projectile.Hitbox.Intersects(heartTarget.Hitbox))
+			{
+				heartTarget.TurnToAir();
+				heartTarget.active = false;
+				RageTime = 600;
+				EatTime = 120;
+				State = 0;
+			}
+		}
+
+		/// <summary>
+		/// The minion attempts to attack an NPC target to deal damage to it
+		/// </summary>
+		public void AttackTarget()
+		{
+			// Abort if target is lost
+			if (npcTarget is null || !npcTarget.active)
+			{
+				npcTarget ??= null;
+				State = 0;
+				return;
+			}
+
+			Projectile.position += (npcTarget.Center - Projectile.Center) * 0.04f;
+
+			Projectile.position += Projectile.Center.DirectionTo(Owner.Center) * (float)Math.Sin(Projectile.timeLeft * 0.1f + Projectile.minionPos) * 2.5f;
+		}
+
+		/// <summary>
+		/// The minion has no target and simply hangs around it's owner
+		/// </summary>
+		public void Idle()
+		{
+			float totalCount = Owner.ownedProjectileCounts[Type] > 0 ? Owner.ownedProjectileCounts[Type] : 1;
+
+			Vector2 idlePos = Owner.Center + new Vector2(-32 + Projectile.minionPos / totalCount * 64, -72);
+			idlePos += Vector2.UnitX.RotatedBy(Projectile.timeLeft / 40f + Projectile.minionPos) * 10;
+
+			Projectile.position += (idlePos - Projectile.Center) * 0.06f;
+
+			Projectile.velocity *= 0.98f;
+		}
+
+		public override void ModifyHitNPC(NPC target, ref NPC.HitModifiers modifiers)
+		{
+			if (Raged)
+				modifiers.SourceDamage += 0.5f;
+		}
+
+		public override void OnHitNPC(NPC target, NPC.HitInfo hit, int damageDone)
```

Comment:
onhitNPC is only executed by the owner of the projectile. you probably want to drop a 

Main.player[Projectile.owner].TryGetModPlayer<StarlightPlayer>(out StarlightPlayer starlightPlayer);
starlightPlayer.SetHitPacketStatus(shouldRunProjMethods: true);

in here (like starwood scepter) to leverage our custom hit packet logic so this can be executed by all clients and the server. Mostly for that newitem since only the server can create new items that way